internal/compiler/emitter.go

```diff
@@ -56,11 +57,11 @@ func (e *emitter) getDeclarationTransformers(emitContext *printer.EmitContext, d
 	return []*declarations.DeclarationTransformer{transform}
 }
 
-func getModuleTransformer(emitContext *printer.EmitContext, options *core.CompilerOptions, resolver binder.ReferenceResolver, getEmitModuleFormatOfFile func(file ast.HasFileName) core.ModuleKind) *transformers.Transformer {
-	switch options.GetEmitModuleKind() {
+func getModuleTransformer(ctx context.Context, resolver binder.ReferenceResolver, getEmitModuleFormatOfFile func(file ast.HasFileName) core.ModuleKind) *transformers.Transformer {
+	switch transformers.GetCompilerOptionsFromContext(ctx).GetEmitModuleKind() {
 	case core.ModuleKindPreserve:
 		// `ESModuleTransformer` contains logic for preserving CJS input syntax in `--module preserve`
-		return moduletransforms.NewESModuleTransformer(emitContext, options, resolver, getEmitModuleFormatOfFile)
+		return moduletransforms.NewESModuleTransformer(ctx, resolver, getEmitModuleFormatOfFile)
```

Comment:
Reading further, I'm surprised to see what is basically a constructor function taking in a context; that usually implies that something is being closed over out of the context and then returned (outside the implied lifetime of the context as a function scoped var), which makes me feel a bit unsure.

Reply:
The other option would be having a struct like `TransformOptions` that just has a grabbag of stuff some transforms use (because we'd like them to have the same constructor shape so they dynamically compose); but the `Context` itself is good for the `EmitResolver` users in particular, since it'll allow us to properly thread the context into those checker API calls without storing it in the host. Not that I've done that yet.